contents/data/nosql/NoSQL.md

```diff
@@ -0,0 +1,437 @@
+and learn to set up a local instance of mongoDB.
+
+Thereafter, you should use a <trigger for="local-shell-advise">**local** _mongo shell_</trigger> to get familiar with mongoDB syntax.<br>
+You can follow the instructions [here](https://docs.mongodb.com/manual/mongo/) to connect to your 
+mongoDB instance from the shell as you had configured earlier.
+
+<popover header="Online playgrounds" id="local-shell-advise">
+<template slot="content">
+
+There are also many online playgrounds that allow you to experiment with mongoDB queries without setting up a
+local database instance and shell, such as [this](https://mongoplayground.net/).
+
+<div>
+However, to put what you've learnt into practice (bulding an application) later, I highly recommend getting your feet wet with the shell and local mongoDB instance first, since it will be necessary to set up your application drivers later on!
+</div>
+</template>
+</popover>
+
+**Basics**
+
+Here are some great resources on mongoDB:
+- [Key Components of MongoDB Architecture](https://www.guru99.com/what-is-mongodb.html) heading here for a quick refresher of key terminology in mongoDB.
+- [Data-flair](https://data-flair.training/blogs/mongodb-create-database/) is a great starting point on <tooltip content="creating databases, collections, etc.">administration</tooltip> of
+your local mongoDB instance. It also provides a higher level overview of each topic than the MongoDB documentation.
+- [MongoDB documentation](https://docs.mongodb.com/manual/crud/) can be overwhelming, but it is also
+a great starting point to learn and test features of mongoDB, and is the defacto reference for it.
+
+To guide you through your journey, here are the essentials that you should go through on the above sites _in order_.
+
+1. Basic database administration
+2. CRUD operations
+3. Data aggregation
+
+**Practice**
+
+After learning these core features and getting familiar with the syntax, you could try your hands at building
+a simple project to get a good feel for NoSQL in an actual backend.
+
+Depending on the backend language you are using, you should browse through the documentation [here](https://docs.mongodb.com/ecosystem/drivers/) for the <trigger for="which-driver">appropriate language</trigger> and learn how to connect to your
+mongoDB instance from your application and utilize the features you learnt above.
+
+<popover header="Which driver should I try first?" id="which-driver">
+<template slot="content">
+
+- Syntax for the different drivers will inevitably vary slightly from language to language. However, the core concepts stay the same.
+- If you want something familiar, and you have knowledge of nodeJS, I highly recommend getting started with the
+nodeJS driver which is very close to the shell syntax.
+</template>
+</popover>
+
+**Advanced**
+
+Congratulations! You should have a good handle on mongoDB by now, and learnt how you would generally interact with NoSQL databases.
```

Comment:
This phrasing implies that the reader has come back to the article after learning MongoDB, which is unlikely.